core/src/main/java/com/tweet/core/model/UserId.java

```diff
@@ -1,13 +1,15 @@
 package com.tweet.core.model;
 
 import javax.persistence.Entity;
+import javax.persistence.Id;
 
 @Entity
-public class User {
+public class UserId {
```

Comment:
User class for fine, I refereed to adding Id field -> long userId :)
With same annotations as in previous entity
@id
@GeneratedValue(strategy = GenerationType.AUTO)
private long userId

Reply:
basically each entity need to have id, other way it wont run. There are few types of ids, you may check it out in free time, they are mainly controlled via @GeneratedValue strategy. We want to stick to AUTO as its best in 90% cases.

Reply:
I plan to use username as userid

Reply:
We totally should use long userId. :)

The main advantage of a surrogate primary key (or technical ID) is that it is one simple number and not a combination of multiple attributes as most natural keys. All involved systems, mainly Hibernate and the database, can handle it very efficiently.

Hibernate also supports ids like username here - but its mainly done to allow migration of legacy system to hibernate (those types of ids were much more popular in past)

You are right that there is a good point to make username unique though ,we can easily achieve that by adding.
@Column(unique=true)
String username;